### PR TITLE
added 'stack' completions

### PR DESCRIPTION
**Pull request by ad0**
Added the completion file for the haskell 'stack' build tool.

### CODE REVIEW

File: share/completions/stack.fish [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+complete -c stack -l no-terminal -d 'Disable overriding terminal detection in the case of running in a false terminal'
+complete -c stack -l stack-yaml -a '(__fish_complete_path)' -d 'Override project stack.yaml file (overrides any STACK_YAML environment variable)'
+
+#
+# Commands:
+#
+
+complete -c stack -a build --description 'Build the package(s) in this directory/configuration'
+complete -c stack -a install --description 'Shortcut for \'build --copy-bins\''
+complete -c stack -a uninstall --description 'DEPRECATED: This command performs no actions, and is present for documentation only'
```
Comment (faho) [inline share/completions/stack.fish]: If it's a deprecated no-op, there's no need to complete it.

**Comment by faho** [comment]
I'm assuming this is [this stack](http://docs.haskellstack.org/en/stable/README.html) a build tool for haskell?

If so, is there any man-page like short overview for all commands and options?

On to the completions themselves: These _look_ alright, but I find the real meat is in the next stage of completions - e.g. in `stack build <TAB>`. Is there a way to complete package names there?

Or, if all that is too complicated, wouldn't it be possible to generate completions of at least this level with `stack --help`? The short [output snippet](http://docs.haskellstack.org/en/stable/GUIDE.html#the-build-synonyms) here looks like it might be transformable with e.g. `stack --help | string replace -r " *" \t` (or sed).

**Comment by ad0** [comment]
It is indeed about this haskell build tool.

There is no need to add something other than "--" options after 'build' command (and it is the case for the majority of the commands), the tool operates mainly wrt the current directory.

Yeah you're right, it should be possible to auto-generate those simple completion rules.. I'll get this going.

**Comment by faho** [comment]
>Yeah you're right, it should be possible to auto-generate those simple completion rules.. I'll get this going.

I've been trying, and while it looks about 90% right, the remaining 10% seem impossible - it does too much pretty-printing for that.

In other words: Remove the deprecated command, make the spelling/case for "experimental" consistent (some of them are "EXPERIMENTAL"), preferably add a comment that this is haskell-stack (with a link to the docs?) and not Plan9 stack and I'll merge.

**Comment by faho** [comment]
Merged as 2a4f21c. Thanks!

**Pull request closed without merging**